cordova-lib/src/cordova/build.js

```diff
@@ -17,22 +17,26 @@
     under the License.
 */
 
-var cordovaUtil      = require('./util'),
-    HooksRunner           = require('../hooks/HooksRunner');
+var Q = require('q'),
+    cordovaUtil = require('./util'),
+    HooksRunner = require('../hooks/HooksRunner');
 
 // Returns a promise.
 module.exports = function build(options) {
-    var projectRoot = cordovaUtil.cdProjectRoot();
-    options = cordovaUtil.preProcessOptions(options);
-
-    // fire build hooks
-    var hooksRunner = new HooksRunner(projectRoot);
-    return hooksRunner.fire('before_build', options)
-    .then(function() {
-        return require('./cordova').raw.prepare(options);
-    }).then(function() {
-        return require('./cordova').raw.compile(options);
-    }).then(function() {
-        return hooksRunner.fire('after_build', options);
+    return Q().then(function() {
+        var opts = cordovaUtil.preProcessOptions(options);
+        var projectRoot = cordovaUtil.cdProjectRoot();
+        return [new HooksRunner(projectRoot), opts];
+    })
+    .spread(function (hooksRunner, options) {
```

Comment:
Well, that'd be my preference, as the code ends up much simpler and easy to read, to my mind. Otherwise we're arbitrarily splitting out the synchronous part of the process into its own separate promise, when there's no need to. But ultimately it's up to you :smile:.